### ISSUE TITLE
torch.nn.modules.pooling return_indices and ceil_mode reversed

### ISSUE THREAD

**Issue by domdal** [pytorch/pytorch#79415]
In _MaxPoolNd the order of return_indices and ceil_mode differs from the order in MaxPool1d (and 2d, 3d) 

_MaxPoolNd 
https://github.com/pytorch/pytorch/blob/0e25a9490ba2991ac4a44d132e3656c9254ca453/torch/nn/modules/pooling.py#L18-L20 

MaxPool1d
https://github.com/pytorch/pytorch/blob/0e25a9490ba2991ac4a44d132e3656c9254ca453/torch/nn/modules/pooling.py#L87-L90
MaxPool2d 
https://github.com/pytorch/pytorch/blob/0e25a9490ba2991ac4a44d132e3656c9254ca453/torch/nn/modules/pooling.py#L161-L164
MaxPool3d
https://github.com/pytorch/pytorch/blob/0e25a9490ba2991ac4a44d132e3656c9254ca453/torch/nn/modules/pooling.py#L239-L242

cc @albanD @mruberry @jbschlosser @walterddr @kshitij12345 @saketh-are

**Comment by jbschlosser** [comment]
Duplicate of #62545 - closing this issue and reopening there instead.

TL;DR: we previously tried to fix this but found that it's too much trouble to be worth it. In particular, maintaining BC (especially wrt JIT) is not easily doable
